api/applications/applications_controller_test.go

```diff
@@ -418,6 +418,21 @@ func TestGetApplication_WithEnvironments(t *testing.T) {
 		WithEnvironment("prod", "release").
 		WithEnvironment(anyOrphanedEnvironment, "feature"))
 
+	commonTestUtils.ApplyEnvironment(builders.
```

Comment:
I think you should not need to do this. I think you have misunderstood the builder. I think the application builder should take care of construction of an environment. A tester should not need to create and apply these environments, unless it is needed for testing. Also the opposite should take place when we want to test an RE. That is a environmentBuilder should construct a registration if not provided